main.tf

```diff
@@ -1,6 +1,7 @@
 locals {
-  elastic_ip        = var.publicly_accessible ? aws_eip.default[0].public_ip : null
-  subnet_group_name = var.subnet_ids == null ? "default" : (var.redshift_subnet_group != null ? var.redshift_subnet_group : var.name)
+  create_logging_bucket = var.logging != null && try(var.logging.create_bucket, false) && try(var.logging.log_destination_type, "") == "s3" ? 1 : 0
```

Comment:
Do we need to compute a value here when we have `var.logging.create_bucket`?

Seems we should either use that or computer the value, not both.

Reply:
I wanted the flexibility to create the S3 bucket outside of this module(maybe a common log bucket where redshift will also use that).

Reply:
changed the local variable logic